File: frontend/src/models/ExchangeRate.ts
export interface ExchangeRate {
  pair: string;
  description: string;
  bid: number;
  ask: number;
  high: number;
  low: number;
  variation: number;
  variationPercent: number;
  timestamp: string;
}

export interface CurrencySnapshot {
  rates: ExchangeRate[];
  fetchedAt: Date;
}

File: backend/src/main/kotlin/com/currencydash/domain/model/ExchangeRate.kt
package com.currencydash.domain.model

import java.math.BigDecimal
import java.time.LocalDateTime

data class ExchangeRate(
    val pairCode: String,
    val description: String,
    val bid: BigDecimal,
    val ask: BigDecimal,
    val high: BigDecimal,
    val low: BigDecimal,
    val variation: BigDecimal,
    val variationPercent: BigDecimal,
    val timestamp: LocalDateTime
)
